src/algebra/jordan/special.lean

```diff
@@ -0,0 +1,198 @@
+"symmetrisation" i.e
+```
+a∘b = 1/2(ab+ba).
+```
+When the original multiplication is associative, the symmetrised algebra is a commutative Jordan
+algebra. A commutative Jordan algebra which can be constructed in this way from an associative
+multiplication is said to be a special Jordan algebra.
+
+## Main results
+
+- comm_jordan : The symmeterised algebra arising from an associative algebra is a commutative Jordan
+  algebra.
+
+## Implementation notes
+
+The approach taken here is inspired by algebra.opposites.
+
+## References
+
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+-/
+
+/--
+The symmetrised algebra has the same underlying space as the original algebra.
+-/
+def sym_alg (α : Type*) : Type* := α
+
+postfix `ˢʸᵐ`:std.prec.max_plus := sym_alg
+
+namespace sym_alg
+
+variables {α : Type*}
+
+/-- The element of `sym_alg α` that represents `a : α`. -/
+@[pp_nodot]
+def sym : α → αˢʸᵐ := id
+
+/-- The element of `α` represented by `x : αˢʸᵐ`. -/
+@[pp_nodot]
+def unsym : αˢʸᵐ → α := id
+
+instance [inhabited α] : inhabited αˢʸᵐ := ⟨sym (default α)⟩
+
+@[simp] lemma unsym_sym (x : α) : unsym (sym x) = x := rfl
+@[simp] lemma sym_unsym (x : α) : sym (unsym x) = x := rfl
+
+instance [has_zero α] : has_zero (αˢʸᵐ) := { zero := sym 0 }
+instance [has_sub α] : has_sub αˢʸᵐ := { sub := λ a b, sym (unsym a - unsym b) }
+
+instance [has_add α] : has_add αˢʸᵐ :=
+{ add := λ a b, sym (unsym a + unsym b) }
+
+@[simp] lemma sym_add [has_add α] (a b : α) : sym (a + b) = sym a + sym b := rfl
+@[simp] lemma unsym_add [has_add α] (x y : αˢʸᵐ) : unsym (x + y) = unsym x + unsym y := rfl
+
+@[simp] lemma sym_sub [has_sub α] (x y : α) : sym (x - y) = sym x - sym y := rfl
+@[simp] lemma unsym_sub [has_sub α] (x y : αᵐᵒᵖ) : unsym (x - y) = unsym x - unsym y := rfl
+
+instance [has_neg α] : has_neg αˢʸᵐ :=
+{ neg := λ a, sym (-unsym a) }
+
+instance {R : Type*} [has_scalar R α] : has_scalar R αˢʸᵐ :=
+{ smul := λ r a, sym (r • unsym a) }
+
+@[simp] lemma sym_smul {R : Type*} [has_scalar R α] (c : R) (a : α) : sym (c • a) = c • sym a := rfl
+
+lemma unsym_injective : function.injective (@unsym α) := λ _ _, id
+
+instance [add_comm_semigroup α] : add_comm_semigroup (αˢʸᵐ) :=
+unsym_injective.add_comm_semigroup _ (λ _ _, rfl)
+
+instance [add_monoid α] : add_monoid (αˢʸᵐ) :=
+unsym_injective.add_monoid_smul _ rfl (λ _ _, rfl) (λ _ _, rfl)
+
+instance [add_group α] : add_group (αˢʸᵐ) :=
+unsym_injective.add_group_smul _ rfl
+  (λ _ _, rfl) (λ _, rfl) (λ _ _, rfl) (λ _ _, rfl) (λ _ _, rfl)
+
+instance [add_comm_monoid α] : add_comm_monoid (αˢʸᵐ) :=
+{ ..sym_alg.add_comm_semigroup, ..sym_alg.add_monoid }
+
+instance [add_comm_group α] : add_comm_group (αˢʸᵐ) :=
+{ ..sym_alg.add_comm_monoid, ..sym_alg.add_group }
+
+
+
+@[simp] lemma sym_zero [has_zero α] : sym (0 : α) = 0 := rfl
+@[simp] lemma unsym_zero [has_zero α] : unsym (0 : αˢʸᵐ) = 0 := rfl
```

Comment:
It would make more sense to me to put these up near `sym_add` and `sym_sub`

Reply:
Done.